tests/actions/ReportTest.js

```diff
@@ -0,0 +1,102 @@
+PushNotification.deregister = () => {};
+
+describe('actions/Report', () => {
+    it('should store a new report action in Onyx when one is handled via Pusher', () => {
+        const TEST_USER_ACCOUNT_ID = 1;
+        const TEST_USER_LOGIN = 'test@test.com';
+        const REPORT_ID = 1;
+        const ACTION_ID = 1;
+        const REPORT_ACTION = {
+            actionName: 'ADDCOMMENT',
+            actorAccountID: TEST_USER_ACCOUNT_ID,
+            actorEmail: TEST_USER_LOGIN,
+            automatic: false,
+            avatar: 'https://d2k5nsl2zxldvw.cloudfront.net/images/avatars/avatar_3.png',
+            isAttachment: false,
+            isFirstItem: false,
+            loading: true,
+            message: [{type: 'COMMENT', html: 'Testing a comment', text: 'Testing a comment'}],
+            person: [{type: 'TEXT', style: 'strong', text: 'Test User'}],
+            sequenceNumber: ACTION_ID,
+            shouldShow: true,
+            timestamp: moment().unix(),
+        };
+
+        // When using the Pusher mock the act of calling Pusher.isSubscribed will create a
+        // channel already in a subscribed state. These methods are normally used to prevent
+        // duplicated subscriptions, but we don't need them for this test.
+        Pusher.isSubscribed = jest.fn().mockReturnValue(false);
+        Pusher.isAlreadySubscribing = jest.fn().mockReturnValue(false);
+
+        // Connect to Pusher
+        PusherConnectionManager.init();
+        Pusher.init({
+            appKey: CONFIG.PUSHER.APP_KEY,
+            cluster: CONFIG.PUSHER.CLUSTER,
+            authEndpoint: `${CONFIG.EXPENSIFY.URL_API_ROOT}api?command=Push_Authenticate`,
+        });
+
+        let reportActions;
+        Onyx.connect({
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${REPORT_ID}`,
+            callback: val => reportActions = val,
+        });
+
+        // Set up Onyx with some test user data
+        return signInWithTestUser(TEST_USER_ACCOUNT_ID, TEST_USER_LOGIN)
+            .then(() => {
+                subscribeToReportCommentEvents();
+                return waitForPromisesToResolve();
+            })
+            .then(() => fetchPersonalDetailsForTestUser(TEST_USER_ACCOUNT_ID, TEST_USER_LOGIN, {
+                [TEST_USER_LOGIN]: {
+                    accountID: TEST_USER_ACCOUNT_ID,
+                    email: TEST_USER_LOGIN,
+                    firstName: 'Test',
+                    lastName: 'User',
+                },
+            }))
+            .then(() => {
+                // This is a fire and forget response, but one it completes we should be able to verify that we
+                // have an "optimistic" report action in Onyx.
+                addAction(REPORT_ID, 'Testing a comment');
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                const resultAction = reportActions[ACTION_ID];
+                expect(resultAction).toEqual(REPORT_ACTION);
```

Comment:
Adding this assertion will help identify that yes, this is definitely an optimistic comment, and not the final result.

```
expect(resultAction.loading).toEqual(true);
```

This, of course, is already implied in the current assertion, but it's an extra mental jump that has to be made which isn't totally obvious.

Reply:
Yep that makes sense to me. I think it'll be a more interesting test when the `clientID` stuff comes along. But this is mostly what we are looking for now and proves that the action was handled from the Pusher payload.

Reply:
Yeah, this will look cool once the clientID stuff happens, and it will be a perfect use for these tests!